### PR TITLE
OCM-2690 | feat: Support HTPasswd from-file flag in interactive mode

### PR DESCRIPTION
**Pull request by ckandag**
https://issues.redhat.com/browse/OCM-2690

### CODE REVIEW

**Review by ciaranRoche — COMMENTED** [review]
Overall looks good, I agree with Ori's comment, if the flag isnt provided we should prompt for it in interactive mode, if the user is not supplying the other args. 

Also is there anything we can do to the func `buildUserList` the amount of `if elses` can be hard to follow. Is it possible to flatten this out a little bit, by abstracting to other functions and returning early when possible. 

Also i know its not directly related to this change, but i think it would be good to add some unit tests for `parseHtpasswordFile`

File: cmd/create/idp/htpasswd.go [hunk 1]
```diff
@@ -93,11 +93,32 @@ func validateUserArgs(r *rosa.Runtime) {
...
 func buildUserList(cmd *cobra.Command, r *rosa.Runtime) *cmv1.HTPasswdUserListBuilder {
 
 	userList := make(map[string]string)
 	hashed := false
 
+	htpasswdFile := args.htpasswdFile
+
+	//interactive is enabled and  other user info is not set
+	//then prompt for the optional flag 'from-file'
+	if interactive.Enabled() &&
```
[the remark this answers is not in the document]
Comment (ckandag) [reply cmd/create/idp/htpasswd.go:106]: @oriAdler @ciaranRoche :  So the ask is to check for htpasswdFile == ""  before prompting,  correct?  Does the below sound better
`	if interactive.Enabled() &&
		htpasswdFile == "" && len(args.htpasswdUsers) == 0 && args.htpasswdUsername == "" {
`

[the remark this answers is not in the document]
Comment (oriAdler) [reply cmd/create/idp/htpasswd.go:106]: Yes, if the `htpasswdFile` was provided we can use it.
Otherwise, we can prompt the user with a message to enter the value for `htpasswdFile`.

**Comment by ckandag** [comment]
@ciaranRoche @oriAdler  :   Can you PTAL again
- interactively prompt if filepath is not set and if other args are also set
- refactored the code to simplify the if\else structure, 
- added test for htpasswd parser

File: cmd/create/idp/htpasswd_test.go [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+			Entry("Nonexistent File Test",
+				"", nil, true),
+		)
+	})
+
+})
+
+func CreateTmpFile(content string) (*os.File, error) {
+	// Create a temporary file
+	file, errs := ioutil.TempFile("", "temp-*.txt")
```
Comment (oriAdler) [inline cmd/create/idp/htpasswd_test.go:93]:
```suggestion
	file, err := ioutil.TempFile("", "temp-*.txt")
```

**Review by oriAdler — COMMENTED** [review]
Left a small comment, other than that LGTM.
